component/src/main/java/org/wso2/extension/siddhi/io/cdc/source/CDCSource.java

```diff
@@ -0,0 +1,342 @@
+     *                            related utility functions.
+     */
+    @Override
+    public void init(SourceEventListener sourceEventListener, OptionHolder optionHolder,
+                     String[] requestedTransportPropertyNames, ConfigReader configReader,
+                     SiddhiAppContext siddhiAppContext) {
+        String siddhiAppName = siddhiAppContext.getName();
+        String streamName = sourceEventListener.getStreamDefinition().getId();
+
+        //initialize mandatory parameters
+        String url = optionHolder.validateAndGetOption(CDCSourceConstants.DATABASE_CONNECTION_URL).getValue();
+        String tableName = optionHolder.validateAndGetOption(CDCSourceConstants.TABLE_NAME).getValue();
+        String username = optionHolder.validateAndGetOption(CDCSourceConstants.USERNAME).getValue();
+        String password = optionHolder.validateAndGetOption(CDCSourceConstants.PASSWORD).getValue();
+        operation = optionHolder.validateAndGetOption(CDCSourceConstants.OPERATION).getValue();
+
+        //initialize optional parameters
+        int serverID;
+        serverID = Integer.parseInt(optionHolder.validateAndGetStaticValue(CDCSourceConstants.DATABASE_SERVER_ID,
+                "-1"));
+
+        String serverName;
+        serverName = optionHolder.validateAndGetStaticValue(CDCSourceConstants.DATABASE_SERVER_NAME,
+                CDCSourceConstants.EMPTY_STRING);
+
+        //initialize parameters from connector.properties
+        String connectorProperties = optionHolder.validateAndGetStaticValue(CDCSourceConstants.CONNECTOR_PROPERTIES,
+                CDCSourceConstants.EMPTY_STRING);
+
+        //initialize history file directory
+        carbonHome = CDCSourceUtil.getCarbonHome();
+        historyFileDirectory = carbonHome + File.separator + "cdc" + File.separator + "history"
+                + File.separator + siddhiAppName + File.separator;
+
+        validateParameter();
+
+        //send this object reference and preferred operation to changeDataCapture object
+        changeDataCapture = new ChangeDataCapture(operation, sourceEventListener);
+
+        //create the folder for history file if not exists
+        File directory = new File(historyFileDirectory);
+        if (!directory.exists()) {
+            boolean isDirectoryCreated = directory.mkdirs();
+            if (isDirectoryCreated && log.isDebugEnabled()) {
+                log.debug("Directory created for history file.");
+            }
+        }
+
+        try {
+            Map<String, Object> configMap = CDCSourceUtil.getConfigMap(username, password, url, tableName,
+                    historyFileDirectory, siddhiAppName, streamName, serverID, serverName, connectorProperties,
+                    this.hashCode());
+            changeDataCapture.setConfig(configMap);
+        } catch (WrongConfigurationException ex) {
+            throw new SiddhiAppCreationException("The cdc source couldn't get started. Invalid" +
```

Comment:
When logging errros print what do you expect and what you found. The CDC source couldt start because of invalid configurations, found configurations, .....

Reply:
Fixed with commit: `Fix some suggested issues in PR#1.` 8a89f22